src/plugins/intel_gpu/src/graph/impls/cpu/non_max_suppression.cpp

```diff
@@ -403,11 +410,23 @@ struct non_max_suppression_impl : typed_primitive_impl<non_max_suppression> {
 namespace detail {
 
 attach_non_max_suppression_impl::attach_non_max_suppression_impl() {
-    implementation_map<non_max_suppression>::add(impl_types::cpu, non_max_suppression_impl::create, {
-        std::make_tuple(data_types::i32, format::bfyx),
-        std::make_tuple(data_types::f16, format::bfyx),
-        std::make_tuple(data_types::f32, format::bfyx),
-    });
+    implementation_map<non_max_suppression>::add(impl_types::cpu,
+                                                 non_max_suppression_impl::create,
+                                                 {
+                                                     std::make_tuple(data_types::i32, format::bfyx),
+
+                                                     std::make_tuple(data_types::f16, format::bfyx),
+                                                     std::make_tuple(data_types::f16, format::b_fs_yx_fsv16),
+                                                     std::make_tuple(data_types::f16, format::bs_fs_yx_bsv16_fsv16),
```

Comment:
I dont think cpu impl is supporting blocked format... It is loading data from continuous memory.
Is this validated? 
If not, I think we should do like this :
In get_preferred_format for nms operation in layout_optimizer, if preferred impl is cpu => set the expected format as bfyx, and if preferred_impl is ocl, set as format::any.

Reply:
I'm working on it now. Tests are passed but only because we have 'reorder' layer removal on ngraph optimization level. So in fact we receive plain format in input to opreration. Investigating how I can avoid this removal.

Potentially I can prepare data in separate network for tests and push it to non_max_suppression primitive in separate network to avoid this reorder layer removal, but I think it is in general incorrect behaviour when reorder which set data to another format is removed in optimizer, how do you think?

Reply:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/plugin/ops/non_max_suppression.cpp You mean this reorder, right? Yeah I think we should get rid of this reorder. If reorder is needed, it should be added by get_preferred_format so that the required reorder can be inserted in reorder_input pass.

Reply:
Yes I mean reorder which converts from plain to blocked format
And it is removed here:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp#L144

Reply:
The liknked page is merging two redundant reorders to one reorder.

Reply:
but finally to my operation node here
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/graph/impls/ocl/non_max_suppression.cpp#L57
I receive plain layouts (bfyx) in inputs, but expect to have blocked in dependencies

Reply:
My situation
Input bfyx -> reordered_input fsv
Boxes bfyx-> reordered_boxes fsv
reordered_input, reordered_boxes -> nonmaxsuppression

So I expext to have fsv as inputs to nonmaxsuppression but because of optimization I receive bfyx in dependencies

Reply:
> I dont think cpu impl is supporting blocked format... It is loading data from continuous memory. Is this validated? If not, I think we should do like this : In get_preferred_format for nms operation in layout_optimizer, if preferred impl is cpu => set the expected format as bfyx, and if preferred_impl is ocl, set as format::any.

could you check if my understanding to provide implementation is correct
https://github.com/openvinotoolkit/openvino/pull/11890/commits/e48fd96eb05e3572747f1ae3a8c294854dc7a54f#diff-41a32bc38b6fa9c7db5e656050c380c993d424df879bb6a1e2627e7008a7704bR1407

Reply:
Ah you set the impl to be ocl if it is blocked format? That make sense.

Reply:
About the unexpected reorder removal, if I understand correctly that you want to keep the blocked format for the unittest only, could you try this?
     options_unfused.set_option(cldnn::build_option::optimize_data(false));
     options_unfused.set_option(cldnn::build_option::allow_static_input_reorder(true));

Reply:
I've already handled this - tests now are working as expected and testing the blocked format. The issue was that all inputs and outputs should be blocked. If even only one is plane - reorder will be added to make all the inputs plane